src/Symfony/Component/HttpKernel/HttpKernel.php

```diff
@@ -229,7 +229,7 @@ private function handleException(\Exception $e, $request, $type)
         if (!$event->hasResponse()) {
             $this->finishRequest($request, $type);
 
-            throw $e;
+            throw new \LogicException('Handling "kernel.exception" events should led to a Response', 0, $e);
```

Comment:
I reworded the message to `No listeners of the "kernel.exception" event set a Response`